### PR TITLE
Add bag-of-words model and basic training framework

### PR DESCRIPTION
**Pull request by ericagreene**
This change adds a basic framework for training a model in TensorFlow for the toxicity Kaggle competition. 

### What the script does
* loads the training data for the toxicity competition (you'll need to download it from [here](https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge/data) first)
* splits it into a training and held-out test set
* trains a model
* evaluates the model on the held-out data
* makes predictions for the held-out data and the unlabeled test data and writes them to a file

### What the script does not do
* Search over the parameter space
* Save the model
* I make no claims about good accuracy, I haven't optimized for that yet

The model here is just a simple bag of words model with a learned word embedding layer. My plan is to use this code to play around with different simple baselines for the Kaggle competition that are trained separately from the Google ML training infrastructure. _(And to use that task as a way to learn TensorFlow.)_

### To Run

1. Download the training (`train.csv`) and test (`test.csv`) data from the
[Kaggle challenge](https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge/data).
2. Install library dependencies:
```shell
pip install -r requirements.txt
```

3. Run a model on a given class (e.g. 'toxic' or 'obscene'):

```shell
python model.py --train_data=train.csv --predict_data=test.csv --y_class=toxic --model=bag_of_words
```

### Next Steps
I made issues [here](https://github.com/conversationai/wikidetox/issues?q=is%3Aissue+is%3Aopen+label%3Akaggle) for the features I'd like to add. Let me know if you want me to include any or them in this PR.

### CODE REVIEW

File: conversation_classification/kaggle/model.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            .format(train_percent))
+      raise ValueError
+
+    tf.logging.info("Training on class '{}'".format(y_class))
+
+    # Sample the data to create training data
+    data_train = self.data.sample(frac=train_percent, random_state=seed)
+
+    # Use remaining examples as test data
+    data_test = self.data[self.data["id"].isin(data_train["id"]) == False]
```
[the remark this answers is not in the document]
Comment (ericagreene) [reply conversation_classification/kaggle/model.py]: Good call. I didn't know about that function. I'll update this.

File: conversation_classification/kaggle/model.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    tf.logging.info('----------Evaluation on Held-Out Data---------')
+    tf.logging.info('Accuracy (sklearn)\t: {0:f}'.format(sklearn_score))
+    tf.logging.info('Accuracy (tensorflow)\t: {0:f}'.format(tf_scores['accuracy']))
+    tf.logging.info('')
+
+    # If specified, predict on unlabeled data
+    if FLAGS.predict_data is None:
+      return
+
+    data_unlabeled = WikiData(FLAGS.predict_data).data
```
[the remark this answers is not in the document]
Comment (ericagreene) [reply conversation_classification/kaggle/model.py]: I went back and forth on this. I think you're right that now it doesn't add much as is. My thought was that in the future, if I want to do some preprocessing of the comment text and I want guarantee that I do the same thing at train and test time, it would be nice to encapsulate those steps in a class that represents collections of examples of wikipedia data.  I'd like to keep it this way for now, but if I end up not doing any pre-processing, I'll takeout the WikiData class because it is unnecessarily verbose.
